lib/features/base/base_mailbox_controller.dart

```diff
@@ -554,4 +597,18 @@ abstract class BaseMailboxController extends BaseController {
     mailboxNode = personalMailboxTree.value.findNodeOnFirstLevel((node) => node.item.name?.name.toLowerCase() == name);
     return mailboxNode;
   }
+
+  void copySubaddressAction(BuildContext context, String subaddress) {
+    Clipboard.setData(ClipboardData(text: subaddress));
+    appToast.showToastSuccessMessage(
+        currentOverlayContext!,
+        AppLocalizations.of(currentContext!).emailSubaddressCopiedToClipboard);
+  }
+
+  String getSubaddress(String userEmail, String folderName) {
+    int atIndex = userEmail.indexOf('@');
+    String localPart = userEmail.substring(0, atIndex);
+    String domain = userEmail.substring(atIndex + 1);
+    return "$localPart+$folderName@$domain";
```

Comment:
if `atIndex=-1` it may throw an exception. You should write unit test for this function.